src/kontrol/prove.py

```diff
@@ -680,20 +720,37 @@ def _update_cterm_from_node(cterm: CTerm, node: KCFG.Node, contract_name: str) -
     singlecall_cell = node.cterm.cell('SINGLECALL_CELL')
     gas_cell = node.cterm.cell('GAS_CELL')
     callgas_cell = node.cterm.cell('CALLGAS_CELL')
-    new_accounts = [CTerm(account, []) for account in flatten_label('_AccountCellMap_', new_accounts_cell)]
+
+    all_accounts = flatten_label('_AccountCellMap_', new_accounts_cell)
+
+    new_accounts = [CTerm(account, []) for account in all_accounts if (type(account) is KApply and account.is_cell)]
+    non_cell_accounts = [account for account in all_accounts if not (type(account) is KApply and account.is_cell)]
+
     new_accounts_map = {account.cell('ACCTID_CELL'): account for account in new_accounts}
-    test_contract_account = new_accounts_map[Foundry.address_TEST_CONTRACT()]
-
-    new_accounts_map[Foundry.address_TEST_CONTRACT()] = CTerm(
-        set_cell(
-            test_contract_account.config,
-            'CODE_CELL',
-            KEVM.bin_runtime(KApply(f'contract_{contract_name}')),
-        ),
-        [],
-    )
 
-    new_accounts_cell = KEVM.accounts([account.config for account in new_accounts_map.values()])
+    if Foundry.address_TEST_CONTRACT() in new_accounts_map:
+        test_contract_account = new_accounts_map[Foundry.address_TEST_CONTRACT()]
+
+        new_accounts_map[Foundry.address_TEST_CONTRACT()] = CTerm(
+            set_cell(
+                test_contract_account.config,
+                'CODE_CELL',
+                KEVM.bin_runtime(KApply(f'contract_{contract_name}')),
+            ),
+            [],
+        )
```

Comment:
I think maybe this whole block can be removed, since we now insert the deployed bytecode during constructor execution to begin with, and the deployed bytecode should be the same between `setUp` function and test function execution.